### PR TITLE
Fix FunctionExecutor null ref

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-functions-host/issues/2785

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs [hunk 1]
```diff
@@ -133,8 +132,11 @@ public HostOutputMessage HostOutputMessage
                     logCompletedCancellationToken = cancellationToken;
                 }
 
-                await NotifyCompleteAsync(instanceLogEntry, functionCompletedMessage.Arguments, exceptionInfo);
-                _resultsLogger?.LogFunctionResult(instanceLogEntry);
+                if (instanceLogEntry != null)
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs]: Avoid operating on instanceLogEntry if it is null

Comment (brettsam) [reply src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs]: Why is this ever null? That would mean we're missing some important details about the function executing.

Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs]: See above - if NotifyPreBind fails it can be null.

Comment (brettsam) [reply src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs]: What exception does the user see now?

Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs]: "Unable to resolve function name 'xxx'." See the repro test case I added in https://github.com/Azure/azure-functions-host/pull/3094

Comment (brettsam) [reply src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs]: Ok, cool. I just wanted to make sure we weren't running real functions that were having these calls to `Notify***` skipped.

**Review by brettsam — APPROVED** [review]

**Pull request merged**